src/jobs/calculateRewards.ts

```diff
@@ -0,0 +1,328 @@
+
+        if (rewards.length > 0) {
+            // call the publish rewards method to push the rewards to the smart contract
+            await publishRewards(feeCollector, address, rewards, MAIN_TRANSFER_ADDRESS);
+            await RewardsService.addRewards(address, new Date(), rewardEntryToMapping(rewards));
+        }
+    }
+    console.groupEnd();
+}
+
+/**
+ * Method to calculate and store the share per token locally in a map to be stored in the final map later.
+ * @param measurements The measurements from the database grouped by tokens per day f.e [3000, 4000, 5000]
+ * @param dailyBalanceFeeCollector The daily balance that can be distributed among the users PER token (f.e 7 for THX, 7 for DOIS, etc).
+ */
+async function calculateShare(measurements: IMeasurement[], dailyBalanceFeeCollector: BigNumber) {
+    const measurementsPerToken = new Map<IMeasurement, Map<string, number[]>>();
+    const totalRewardsPerToken = new Map<string, BigNumber>();
+
+    // foreach measurement set median
+    for (const measurement of measurements) {
+        measurementsPerToken.set(measurement, getValuesFromObject(measurement.tokens));
+    }
+
+    for (const [measurement, data] of measurementsPerToken) {
+        const totalMedianPerDay = await getTotalMedianDay(measurement._id);
+        for (const [token, values] of data) {
+            // call the algorithm calculation and store the share
+            const share = calculation(values, totalMedianPerDay.get(token), dailyBalanceFeeCollector);
+
+            // when there is no token in the total rewards yet add the token with 0 as value
+            if (totalRewardsPerToken.get(token) == undefined) {
+                totalRewardsPerToken.set(token, new BigNumber(0));
+            }
+            // add the share to the token in the map
+            totalRewardsPerToken.set(token, totalRewardsPerToken.get(token).plus(share));
+        }
+    }
+
+    return totalRewardsPerToken;
+}
+
+/**
+ * Helper function to run the algorithm calculation.
+ */
+export function calculation(values: number[], total: number, dailyBalanceFeeCollector: BigNumber) {
+    const medianMeasurement = new BigNumber(median(values));
+    const totalMedian = new BigNumber(total);
+
+    // calculate the share which is ((median / total median) * 100) * (dailyFeeBalance / 100)
+    const leftSide = new BigNumber(medianMeasurement).div(totalMedian).multipliedBy(100);
+    const rightSide = dailyBalanceFeeCollector.div(100);
+
+    return leftSide.multipliedBy(rightSide);
+}
+
+/**
+ * Method to publish the rewards to the smart contract based on the token and address of the account address.
+ * @param contract The fee collector contract.
+ * @param address The address of the account that the rewards will be awarded to.
+ * @param rewards
+ * @param mainAddress
+ */
+async function publishRewards(contract: Contract, address: string, rewards: RewardEntry[], mainAddress: string) {
+    const temp = rewards.map((e) => ({
```

Comment:
You might want to rename the const, temp isn't really maintainable as it is not self-explanatory.